src/common/providers/https.ts

```diff
@@ -527,12 +584,18 @@ async function checkTokens(
           'Cannot validate AppCheck token. Please update Firebase Admin SDK to >= v9.8.0'
         );
       }
-      const appCheckToken = await apps()
-        .admin.appCheck()
-        .verifyToken(appCheck);
+      let appCheckData;
+      if (skipTokenVerify) {
+        const decodedToken = unsafeDecodeAppCheckToken(appCheck);
+        appCheckData = { appId: decodedToken.app_id, token: decodedToken };
+      } else {
+        appCheckData = await apps()
+          .admin.appCheck()
+          .verifyToken(appCheck);
+      }
       ctx.app = {
-        appId: appCheckToken.appId,
-        token: appCheckToken.token,
+        appId: appCheckData.appId,
```

Comment:
I feel like I'm misreading something here. Why are we using `appCheckData` as a temporary instead of just saving to `ctx.app`?

Reply:
good point!